### PR TITLE
Add API to create user in fineract

### PR DESCRIPTION
**Pull request by sagarrakshe**

### CODE REVIEW

File: user_client.go [hunk 1]
```diff
@@ -12,6 +14,56 @@ type UserInfoResponse struct {
...
+func (client *Client) CreateClient(req *UserCreateRequest) (*UserCreateResponse, error) {
+	tempPath, _ := url.Parse(clientsURL())
+	path := client.HostName.ResolveReference(tempPath).String()
+
+	var response *UserCreateResponse
+
+	// Store phone number in "<country-code>_<phone_number>"
+	req.MobileNo = fmt.Sprintf("%s_%s", req.CountryCode, req.PhoneNumber)
+	// Set these to empty strings so on marshalling these will be ignored, as this endpoint doesn't accept these parameters
+	req.PhoneNumber = ""
```
Comment (poorva29) [inline user_client.go]: Can we set this as `omitempty` when we marshall?

File: user_client.go [hunk 2]
```diff
@@ -22,5 +74,14 @@ func (client *Client) GetClientInfo(clientId string) (*UserInfoResponse, error)
 	if err := client.MakeRequest(http.MethodGet, path, nil, &response); err != nil {
 		return nil, err
 	}
-	return response, nil
+
+	//NOTE: For backward compatibility error is not being raised here, since many numbers are stored without country-code
```
Comment (sapanakothari) [inline user_client.go]: we are only allowing customers from specific countries, so the country code should be known. we cannot send top without having this info

Comment (sagarrakshe) [reply user_client.go]: Here, I don't need to know what is the country code. so basically the format is `<country-code>_<mobile>`, so I don't care what the code is. All I want is I am able to store the country-code and retrieve is back what was stored.

File: user_client_test.go [hunk 3]
```diff
@@ -11,15 +11,25 @@ func TestUserInfo(t *testing.T) {
...
 	clientId := "337"
-	resp, err := client.GetClientInfo(clientId)
-	if err != nil {
-		log.Println(err)
-	}
 
-	log.Println(resp)
+	t.Run("Test to create new user", func(t *testing.T) {
+		req := UserCreateRequest{}
+		resp, err := client.CreateClient(&req)
```
Comment (sapanakothari) [inline user_client_test.go]: where are the client attributes - name etc being stored? this test cases would always fail?

File: user_client_test.go [hunk 4]
```diff
@@ -11,15 +11,25 @@ func TestUserInfo(t *testing.T) {
...
+
+		log.Println(resp)
+	})
+
+	t.Run("Test to fetch the created user", func(t *testing.T) {
+		resp, err := client.GetClientInfo(clientId)
+		if err != nil {
+			t.Fatal(err)
+		}
+
```
Comment (sapanakothari) [inline user_client_test.go]: pls add asserts to verify response object contains important fields in GetCLientInfo

Comment (sapanakothari) [reply user_client_test.go]: also since create CreateCLient, get should be GetCLient

**Pull request closed without merging**